cmd/dump.go

```diff
@@ -302,13 +337,50 @@ func runDump(ctx *cli.Context) error {
 		}
 
 		excludes = append(excludes, setting.RepoRootPath)
-		excludes = append(excludes, setting.LFS.ContentPath)
+		excludes = append(excludes, setting.LFS.Path)
+		excludes = append(excludes, setting.Attachment.Path)
 		excludes = append(excludes, setting.LogRootPath)
 		if err := addRecursiveExclude(w, "data", setting.AppDataPath, excludes, verbose); err != nil {
 			fatal("Failed to include data directory: %v", err)
 		}
 	}
 
+	// attachment default should be under data directory but not always there.
+	switch setting.Attachment.Storage.Type {
+	case setting.LocalStorageType:
+		if _, err := os.Stat(setting.Attachment.Path); !os.IsNotExist(err) {
+			log.Info("Dumping attachment... %s", setting.Attachment.Path)
+			if err := addRecursive(w, "attachments", setting.Attachment.Path, verbose); err != nil {
```

Comment:
addRecursive doesn't really appear to save us much compared to just iterating across the attachments.

Reply:
See https://github.com/lunny/gitea/pull/7/commits/fe22860c7dcb7e59ec9d98cf6895c43e08fce7c1